pipeline/e2e-job-per-environment.yaml

```diff
@@ -3,30 +3,63 @@
 
 parameters:
     jobNameSuffix: ''
-    windowsImage: 'windows-2019'
+    windowsImage: 'windows-2022-secure'
     macImage: 'macOS-11'
-    linuxImage: 'ubuntu-20.04'
+    linuxImage: 'ubuntu-22.04-secure'
 
 jobs:
     - job: 'e2e_tests_windows${{ parameters.jobNameSuffix }}'
+      templateContext:
+          outputs:
+              - output: pipelineArtifact
+                targetPath: '$(System.DefaultWorkingDirectory)/test-results/e2e'
+                artifactName: '$(Agent.JobName)-screenshots-and-extended-logs'
+                condition: failed() # because the detailed chrome logs are ~150MB and uploading them takes ~30s
+                continueOnError: true
+                displayName: publish e2e screenshots and extended logs
+                timeoutInMinutes: 5
```

Comment:
Is there a way to have this in a single location, instead of in 3 placed?

Reply:
Removed the repeated task from this file and moved it to common file used in step